blogposts/2022/notebooks-ci.md

```diff
@@ -0,0 +1,93 @@
+
+Customizability has advantages and disadvantages: the major advantage is that you can build precisely what you want and the major disadvantage is that the more unique something is, the harder you have to work to explain it.
+
+Crossing that gap was the major use case that drove me to adopt Notebooks and make it so important to my workflow.
+
+## Notebooks augment documentation with live code examples
+
+Documentation can easily go stale and lag behind the code it documents, especially for internal tools and libraries where APIs can change liberally. Example code is especially at risk unless special care is taken to ensure documentation is updated in lock-step with your code.
+
+With Notebooks, you can embed code directly into your documentation (and other internal knowledge management systems) and run Sourcegraph searches without having to leave whatever page you’re on. Notebooks augment our existing documentation, allowing developers to read static, explanatory text as well as run searches that return live code results. Developers can read _about_ code and then see the actual code with just a click.
+
+Take a look at the CI docs I contributed to [here](https://docs.sourcegraph.com/dev/background-information/ci), for example. The docs run through explanations of our Buildkite pipelines, pipeline health, pipeline development, Buildkite infrastructure, and how we use GitHub Actions. In the [pipeline development section](https://docs.sourcegraph.com/dev/background-information/ci#pipeline-development), I embedded notebooks that further explain, and demonstrate, the [Run types](https://docs.sourcegraph.com/dev/background-information/ci#run-types) and [Diff types](https://docs.sourcegraph.com/dev/background-information/ci#diff-types) used internally by the pipeline generator.
+
+<video loop autoplay muted playsinline>
+  <source src="https://storage.googleapis.com/sourcegraph-assets/blog/Notebooks/embedded_notebooks.mp4" type="video/mp4" />
+</video>
+
+You can read about these in detail in the docs, but the idea is that the pipeline generator primarily determines what gets run over contributions based on:
+
+1. Run types, which are determined by branch naming conventions, tags, and environment variables; and
+2. Diff types, which are determined by what files have been changed in a given branch.
+
+I created notebooks for Run types and Diff types so that Sourcegraph developers can see them in action and in the code. The Notebook for Run types includes both a search that shows Run types in the codebase and a search for RunTypeMatcher, which further shows how each RunType declares the conditions under which it should be used.
```

Comment:
```suggestion
I created notebooks for Run types and Diff types so that Sourcegraph developers can see them in action and in the code. The notebook for Run types includes both a search that shows Run types in the codebase and a search for RunTypeMatcher, which further shows how each RunType declares the conditions under which it should be used.
```